### ISSUE TITLE
[Bug]: Missing focus ring

### ISSUE THREAD

**Issue by laurelthorburn** [danny-avila/LibreChat#2866]
### What happened?

Currently, the user is unable to identify what they are focused on within the delete modal as no focus ring is provided on the "cancel" or "delete" buttons.

### Steps to Reproduce

1. Create a chat
2. Begin focus in nav bar
3. Click tab and navigate to the trashcan (delete) icon
4. Select Enter
5. Observe modal and look for focus ring, where is it?
6. Tab around the modal and note that your focused is trapped (yay) but you are unable to see when your focus is on the cancel/delete button... only the exit icon in the top right corner

### Screenshots

![missing-focus-ring](https://github.com/danny-avila/LibreChat/assets/88687282/223c481a-7941-4974-918d-5a0a93239b75)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks! I plan on resolving this soon with a quick fix, but a detailed accessibility review is also planned

**Comment by danny-avila** [comment]
> awesome! Please let me know if I can help support this! 🥳 @danny-avila

Of course, PRs always welcome. Here are my notes from looking into it this morning.

We use a lot of primitives found in `client/src/components/ui` so changes here can affect styling app-wide. In some, very few if not null cases, after my PR, a component may still be passing `className` that removes focus rings (but I can't find any that do).

In some other cases, `tabIndex` is set to `-1`, mainly dropdown/select components, preventing use of tabbing--these could use some investigation.

Lastly, I did notice a minor issue where continuing to tab within the conversation list keeps scrolling the list. There should be some way of tabbing to the `<HeaderOptions />` component.

I am aware that when many conversations are fetched, from infinite scrolling, the application slows down from memory usage increase. This can be mitigated (with virtualization) and/or by trimming the data used to populate the list, which can then be fetched in full when clicked. The solution for this is pretty involved but will need to assess the priority for this.
